src/terms/operators.jl

```diff
@@ -108,7 +108,7 @@ struct NonlocalOperator{T <: Real, PT, DT} <: RealFourierOperator
     D::DT
 end
 function apply!(Hψ, op::NonlocalOperator, ψ)
-    Hψ.fourier .+= op.P * (op.D * (op.P' * ψ.fourier))
+    mul!(Hψ.fourier, op.P, (op.D * (op.P' * ψ.fourier)), 1, 1)
```

Comment:
Is using `1` (integer) type-stable here ?

Reply:
My point is that some form of `one(T)` would be better ...

Reply:
I don't see why not: int is always weaker than floating point types, e.g. 1 + Float16(1.) isa Float16

Reply:
I didn't check type stability. I've seen the same pattern elsewhere in the code though (e.g. https://github.com/JuliaMolSim/DFTK.jl/blob/15b4d41d37204299646ad551d8c6dbd9a98ec85e/src/eigen/lobpcg_hyper_impl.jl#L262C9-L262C31). So if that is not safe, we probably need more care elsewhere too.